user-guide/modules/ROOT/pages/advanced-builds.adoc

```diff
@@ -1,3 +1,247 @@
+
+You can specify which standard library you want to use by setting the appropriate flags in your build system. This is dependent on the compiler you're using.
+
+If you're using the Clang compiler and you want to use `pass:[libc++]`, you can use the `-stdlib=libc++` compiler flag. For example:
+
+[source,txt]
+----
+clang++ -stdlib=libc++ YOUR_FILE.cpp -lboost_system
+----
+
+If you're using GCC and you want to use  `pass:[libstdc++]`, you don't need to do anything specific because this is the default standard library for GCC. If you're using Clang and want to use  `pass:[libstdc++]`, you can use the `-stdlib=libstdc++` flag:
+
+[source,txt]
+----
+clang++ -stdlib=libstdc++ your_file.cpp -lboost_system
+----
+
+When you're building the Boost libraries themselves, you can also specify the standard library to use with the `cxxflags` and `linkflags` options. The following command will build the Boost libraries using Clang with `pass:[libc++]`.
+
+[source,txt]
+----
+./b2 toolset=clang cxxflags="-stdlib=libc++" linkflags="-stdlib=libc++"
+----
+
+
+
+== Specify Static or Dynamic Linking
+
+The compiler options `link=static` and `link=dynamic` specify how the Boost libraries should be linked to your project. If you do not specify a type of linking, `dynamic` is the default.
+
+. The `link=static` option specifies static linking. When you link statically, the linker takes the object code from the library and includes it in the executable file for the program. This leads to larger executables, but the resulting program is self-contained and does not require the library to be present on the system where the program is run. It can be beneficial when you want to avoid dependencies on shared libraries, particularly when you distribute your software to other systems.
++
+[source,txt]
+----
+./b2 link=static
+----
++
+Apart from distributing applications, the other compelling reasons to use static linking are to avoid dependency or version conflicts, and perhaps improved performance, as the application doesn't need to load shared libraries at runtime.
+
+. The `link=dynamic` option specifies dynamic linking, also known as shared linking. With dynamic linking, the linker only includes a small amount of code to locate and interface with the actual library code, which is kept in a separate file. This results in smaller executables and allows for library code to be shared between multiple programs. However, the resulting program requires that the shared library be available at both build time and run time. The build options `dynamic` and `shared` are synonymous.
++
+[source,txt]
+----
+./b2 link=dynamic
+./b2 link=shared
+----
+
+Not all Boost libraries need to be linked to your program. Many are "header-only", meaning you only need to include their headers in your source files and don't need to worry about linking, as there is nothing to link. 
+
+== Build Boost with CMake
```

Comment:
We already have a section about building boost with cmake in the getting started guide. The user should already know that they can build with both b2 and cmake and that b2 is recommended. This section could be about cmake options, but it can't be about presenting the possibility of cmake because that already happened a few pages ago.